File: apps/ios/Barns/BarnsApp/Domain/UseCases/Auth/LoginAsGuestUseCase.swift
/// Signs in with the mock guest account.
struct LoginAsGuestUseCase {
    private let authRepository: AuthRepository

    init(authRepository: AuthRepository) {
        self.authRepository = authRepository
    }

    func execute() async throws -> User {
        try await authRepository.signInWithMockAccount()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/auth/LoginAsGuestUseCase.kt
package com.barns.app.domain.usecase.auth

import com.barns.app.domain.model.User
import com.barns.app.domain.repository.AuthRepository

/**
 * Signs in with the mock guest account.
 */
class LoginAsGuestUseCase(
    private val authRepository: AuthRepository,
) {
    suspend fun execute(): User = authRepository.signInWithMockAccount()
}
